content/tutorials/reactor-firehose-kinesis.textile

```diff
@@ -0,0 +1,602 @@
+
+In this tutorial you will learn how to use the Reactor Firehose integration with Amazon Kinesis. You will publish messages representing random sensor data from a configurable number of simulated IoT devices to a channel and create an integration that automatically pushes these messages into a "Amazon Kinesis Data Stream":https://aws.amazon.com/kinesis/data-streams/.
+
+You will then write a serverless function using "AWS Lambda":https://aws.amazon.com/lambda/ that reads data from the stream and calculates a running average of temperature and humidity readings. It will send these averages as an Ably message to another channel so that a subscriber can display them. This process is illustrated in the following diagram:
+
+<a href="/images/tutorials/reactor-firehose-kinesis/firehose-kinesis-tutorial-overview.png" target="_blank">
+  <img src="/images/tutorials/reactor-firehose-kinesis/firehose-kinesis-tutorial-overview.png" style="width: 100%"
+  title="Click to enlarge" alt="Reactor Firehose with AWS Kinesis tutorial overview">
+</a>
+
+The solution code for this project is available "on Github":https://github.com/ably/firehose-kinesis-tutorial.
+
+h2. Prerequisites
+
+To use *Reactor Firehose*, you need:
+
+* An "Ably account":https://ably.com/sign-up, with the "Enterprise package":https://ably.com/pricing
+
+To use *Amazon Kinesis*, you must:
+
+* Create an AWS account, or use an existing one
+* Create an Administrator IAM user
+* Obtain your access key ID and secret access key
+
+You can achieve these tasks by performing steps 1-3 in "this Amazon guide":https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-setting-up.html. Return here when you are done.
+
+Although all of the AWS configuration steps required here can be performed in the "AWS Console":https://console.aws.amazon.com, this tutorial uses the "AWS CLI":https://aws.amazon.com/cli/. To follow along, first "install the latest version of the AWS CLI":https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-install.html. Then "configure the AWS CLI":https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html#cli-configure-quickstart-config with your IAM user's security credentials.
+
+The code for this tutorial is written in *Node.js*, which you can "download here":https://nodejs.org/.
+
+h2. Step 1: Create your Kinesis data stream
+
+Execute the following @aws@ command in the terminal to create a Kinesis data stream called @sensor-data-stream@, with a single shard:
+
+```[sh]
+aws kinesis create-stream --stream-name sensor-data-stream --shard-count 1
+```
+
+Then, run the following @describe-stream@ command to view details of the stream and confirm that it has been created successfully:
+
+```[sh]
+aws kinesis describe-stream --stream-name sensor-data-stream
+```
+
+Make a note of the @StreamARN@ key's value. You will need it in the next and later steps:
```

Comment:
I'd suggest "future" or "subsequent" steps instead of "next and later". WDYT? 

> You will need it in the next and later steps:

Reply:
Done